packages/lesswrong/components/posts/NewDialogueDialog.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import Button from '@material-ui/core/Button';
+import DialogActions from '@material-ui/core/DialogActions';
+import { useCreate } from '../../lib/crud/withCreate';
+import { useMessages } from '../common/withMessages';
+import { useNavigation } from '../../lib/routeUtil';
+import Input from '@material-ui/core/Input';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  dialog: {
+    padding: 24,
+    paddingBottom: 12,
+    fontFamily: theme.typography.fontFamily,
+    color: theme.palette.text.normal,
+  },
+  inputRow: {
+    marginBottom: 12,
+  },
+  label: {
+    ...theme.typography.body2,
+    display: "inline-block",
+  },
+  titleInput: {
+    ...theme.typography.body1,
+    ...theme.typography.headerStyle,
+    marginTop: 12,
+    width: 400,
+  },
+  usersListEditor: {
+    marginTop: 12,
+    display: "inline-block",
+  },
+  header: {
+    ...theme.typography.headerStyle,
+    marginTop: 0
+  },
+  info: {
+    ...theme.typography.body2,
+    marginRight: 12
+  }
+})
+
+const NewDialogueDialog = ({onClose, classes}: {
+  onClose: ()=>void,
+  classes: ClassesType,
+}) => {
+  const { UsersListEditor, LWDialog, Loading } = Components;
+  const [title, setTitle] = useState("");
+  const {flash} = useMessages();
+  const [participants, setParticipants] = useState<string[]>([]);
+  const {create: createPost, loading, error} = useCreate({ collectionName: "Posts", fragmentName: "PostsEdit" });
+  const { history } = useNavigation();
+  
+  async function createDialogue() {
+    if (loading) return;
+    if (!title.length) {
+      flash("You must choose a title");
+      return;
+    }
+    const createResult = await createPost({
+      data: {
+        title,
+        draft: true,
+        collabEditorDialogue: true,
+        coauthorStatuses: participants.map(userId => ({userId, confirmed: false, requested: false})),
+        shareWithUsers: participants,
```

Comment:
This is safe because `createDialogue` gets recreated (on rerender) each time `participants` are updated via `setParticipants`?

Reply:
Looks like that to me. Seems fine, I think?